apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/TypeTests.cs

```diff
@@ -494,6 +494,87 @@ private void AssertJsonEqual<T>(string expected, string actual) where T: IMessag
+            var options = SpannerConversionOptions.Default;
+            Assert.Equal(expected, SpannerDbType.ArrayOf(SpannerDbType.Float64).ToProtobufValue(input, options));
+        }
+
+        [Fact]
+        public void DeserializeNullableDoubleArrayContainingNull()
+        {
+            var protobuf = new Value { ListValue = new ListValue { Values = { Value.ForNumber(5.5), Value.ForNull(), Value.ForNumber(10.5) } } };
+            var options = SpannerConversionOptions.Default;
+            var expected = new double?[] { 5.5, null, 10.5 };
+            Assert.Equal(expected, SpannerDbType.ArrayOf(SpannerDbType.Float64).ConvertToClrType<double?[]>(protobuf, options));
+        }
+
+        [Fact]
+        public void DeserializeDoubleArrayContainingNull()
+        {
+            var protobuf = new Value { ListValue = new ListValue { Values = { Value.ForNumber(5.5), Value.ForNull(), Value.ForNumber(10.5) } } };
+            var options = SpannerConversionOptions.Default;
+
+            // The null value is converted into 0
+            var expected = new[] { 5.5, 0.0, 10.5 };
+            Assert.Equal(expected, SpannerDbType.ArrayOf(SpannerDbType.Float64).ConvertToClrType<double[]>(protobuf, options));
+        }
+
+        public static IEnumerable<object[]> UseClrDefaultForNulls { get; } = new[]
+        {
+            new object[] { null, SpannerDbType.Int64, typeof(int?) },
+            new object[] { 0, SpannerDbType.Int64, typeof(int) },
+            new object[] { 0L, SpannerDbType.Int64, typeof(long) },
+            new object[] { null, SpannerDbType.String, typeof(string) }
+        };
+
+        [Theory]
+        [MemberData(nameof(UseClrDefaultForNulls))]
+        public void TestNullConversion_LegacyNullHandling(object expectedValue, SpannerDbType dbType, System.Type targetClrType)
+        {
+            var builder = new SpannerConnectionStringBuilder("UseClrDefaultForNull = true");
+            var options = SpannerConversionOptions.ForConnectionStringBuilder(builder);
+            var input = Value.ForNull();
+            var actual = dbType.ConvertToClrType(input, targetClrType, options, topLevel: true);
+            Assert.Equal(expectedValue, actual);
+        }
+
+        [Fact]
+        public void TestNullConversion_DefaultHandling()
+        {
+            var input = Value.ForNull();
+            var options = SpannerConversionOptions.Default;
+            Assert.Throws<InvalidCastException>(() => SpannerDbType.String.ConvertToClrType<string>(input, options));
+            Assert.Throws<InvalidCastException>(() => SpannerDbType.Int64.ConvertToClrType<int>(input, options));
+            Assert.Equal(DBNull.Value, SpannerDbType.String.ConvertToClrType<object>(input, options));
+#if !NETCOREAPP1_0
+            Assert.Equal(DBNull.Value, SpannerDbType.String.ConvertToClrType<IConvertible>(input, options));
```

Comment:
Test `ISerializable` as well?

Reply:
Done. (I don't think it's *particularly* necessary, but it doesn't hurt.)